src/fsharp/TypeChecker.fs

```diff
@@ -4700,7 +4700,14 @@ and TcPatBindingName _cenv _env id ty isMemberThis vis1 topValData (inlineFlag,d
     (fun (TcPatPhase2Input values) -> 
         let (vspec,typeScheme) = 
             match values.TryFind id.idText with
-            | Some x -> x
+            | Some value ->
+                let name = id.idText
+                if not (String.IsNullOrEmpty name) && Char.IsLower(name.[0]) then
```

Comment:
Ok me neither. However it does seem surprising that a identifier is the empty string.